lib/s3routes/routesUtils.ts

```diff
@@ -441,23 +456,38 @@ function retrieveData(
                 });
                 // errors on server side with readable stream
                 readable.on('error', err => {
-                    log.error('error piping data from source');
+                    log.error('error piping data from source', { error: err });
                     _destroyResponse();
                     return cbOnce(err);
                 });
+                // At this point, we assume that the data backend is available.
+                // We write the http headers before piping the data to the
+                // response object.
+                // Any subsequent error leads to destror deletion.
+                if (!response.headersSent) {
```

Comment:
this should always be `false`, right?

Reply:
Yes. As discussed this is more a security (lack of types where we do not have full confidence if we don't have a hidden route that sent the headers already). I can confirm that I see none currently. However, it will be handled in the long term ticket; ARSN-474 that will follow up this ticket with the long term rework.